### PR TITLE
AK+LibJS: Introduce JS::HeapFunction

### PR DESCRIPTION
**Pull request by kalenikaliaksandr**
This change introduces HeapFunction, which is intended to be used as a
replacement for SafeFunction. The new type behaves like a regular
GC-allocated object, which means it needs to be visited from
visit_edges, and unlike SafeFunction, it does not create new roots for
captured parameters.

Co-Authored-By: Andreas Kling <kling@serenityos.org>

### CODE REVIEW

File: AK/Function.h [hunk 1]
```diff
@@ -254,9 +276,11 @@ class Function<Out(In...)> {
         other.m_kind = FunctionKind::NullPointer;
     }
 
+    size_t m_size { 0 };
```
Comment (LucasChollet) [inline AK/Function.h:279]: Can't that think be a compile-time variable?

**Review by LucasChollet — COMMENTED** [review]
Also, we usually use `+` to separate categories in the commit title

File: AK/Function.h (hunk 1 above)
Comment (kalenikaliaksandr) [reply AK/Function.h:279]: I think it can't. `m_size` might change in runtime by `operator=` for example.

Comment (LucasChollet) [reply AK/Function.h:279]: Sure, I didn't think of that case, my bad!

**Pull request merged**